Feature/793 wiring components docs

tomaszguzialek opened the pull request:
Is that good enough? :)

Fixes #793

tomaszguzialek commented:
I gave up with a nice continuous screenshot of the whole page...

1:
![wiring_screen_001](https://cloud.githubusercontent.com/assets/1483194/10542006/540306f0-7415-11e5-820c-357e22b09088.png)

2:
![wiring_screen_002](https://cloud.githubusercontent.com/assets/1483194/10542007/5425729e-7415-11e5-9519-9d821ade4a1e.png)

3:
![wiring_screen_003](https://cloud.githubusercontent.com/assets/1483194/10542008/542678d8-7415-11e5-8dec-e7fc8c433457.png)

kaspersorensen commented:
I think we need to make it clear that there are three ways of wiring and each of them should have it's own little section.

 * Transformer output
 * Filter requirements
 * Output data streams

The text for these sections is almost there in the previous section, we just need to make sub sections

ClaudiaPHI commented on documentation/src/docbkx/chapter12-getting-started-desktop.xml:
```diff
@@ -281,6 +281,48 @@
+				flow". However, some components are able to produce additional
+				output
+				data streams. For example, the main feature of a Completeness
+				Analyzer is to
```
Can you add a link to the "OutputDataStream" documentation ?

ClaudiaPHI replied:
Could you also add an image with "Completeness analyzer" output streams:complete and incomplete ?

tomaszguzialek replied:
AFAIR we don't have any other chapter regarding output data streams. But I will consider a screenshot with Complete records and Incomplete records output data streams and jobs built around them.

tomaszguzialek commented:
Added subsection titles.
Added an example job with output data streams at the end:
![2015-10-19 10_19_01-wiring components together](https://cloud.githubusercontent.com/assets/1483194/10572692/e4137cac-764a-11e5-9956-e00a18dd7e24.png)

ClaudiaPHI commented:
:+1:
